### ISSUE TITLE
2.1.1 windows installed as local user error searching for %programdata%\Git\Config

### ISSUE THREAD

**Issue by lachlan-00** [desktop/desktop#8109]
C:\ProgramData/Git/config doesn't exist. (installed into %appdata% so this folder doesn't exist)

I can't push/pull do anything in my repos now after updating.
![image](https://user-images.githubusercontent.com/1305249/63071105-3dbc6300-bf61-11e9-872b-f55eca33ed8c.png)

Removed entire git folders and config without effect. i'm not an admin on this pc so it can't be installed in programdata.
```
warning: 'C:\ProgramData/Git/config' has a dubious owner: '(unknown)'.
For security reasons, it is therefore ignored.
To fix this, please transfer ownership to an admininstrator.
warning: 'C:\ProgramData/Git/config' has a dubious owner: '(unknown)'.
For security reasons, it is therefore ignored.
To fix this, please transfer ownership to an admininstrator.
remote: Enumerating objects: 2026, done.        
remote: Counting objects:   0% (1/1151)        
remote: Counting objects:   1% (12/1151)        
remote: Counting objects:   2% (24/1151)        
remote: Counting objects:   3% (35/1151)        
remote: Counting objects:   4% (47/1151)        
remote: Counting objects:   5% (58/1151)        
remote: Counting objects:   6% (70/1151)        
remote: Counting objects:   7% (81/1151)        
remote: Counting objects:   8% (93/1151)        
remote: Counting objects:   9% (104/1151)        
remote: Counting objects:  10% (116/1151)        
remote: Counting objects:  11% (127/1151)        
remote: Counting objects:  12% (139/1151)        
remote: Counting objects:  13% (150/1151)        
remote: Counting objects:  14% (162/1151)        
remote: Counting objects:  15% (173/1151)        
remote: Counting objects:  16% (185/1151)        
remote: Counting objects:  17% (196/1151)        
remote: Counting objects:  18% (208/1151)        
remote: Counting objects:  19% (219/1151)        
remote: Counting objects:  20% (231/1151)        
remote: Counting objects:  21% (242/1151)        
remote: Counting objects:  22% (254/1151)        
remote: Counting objects:  23% (265/1151)        
remote: Counting objects:  24% (277/1151)        
remote: Counting objects:  25% (288/1151)        
remote: Counting objects:  26% (300/1151)        
remote: Counting objects:  27% (311/1151)        
remote: Counting objects:  28% (323/1151)        
remote: Counting objects:  29% (334/1151)        
remote: Counting objects:  30% (346/1151)        
remote: Counting objects:  31% (357/1151)        
remote: Counting objects:  32% (369/1151)        
remote: Counting objects:  33% (380/1151)        
remote: Counting objects:  34% (392/1151)        
remote: Counting objects:  35% (403/1151)        
remote: Counting objects:  36% (415/1151)        
remote: Counting objects:  37% (426/1151)        
remote: Counting objects:  38% (438/1151)        
remote: Counting objects:  39% (449/1151)        
remote: Counting objects:  40% (461/1151)        
remote: Counting objects:  41% (472/1151)        
remote: Counting objects:  42% (484/1151)        
remote: Counting objects:  43% (495/1151)        
remote: Counting objects:  44% (507/1151)        
remote: Counting objects:  45% (518/1151)        
remote: Counting objects:  46% (530/1151)        
remote: Counting objects:  47% (541/1151)        
remote: Counting objects:  48% (553/1151)        
remote: Counting objects:  49% (564/1151)        
remote: Counting objects:  50% (576/1151)        
remote: Counting objects:  51% (588/1151)        
remote: Counting objects:  52% (599/1151)        
remote: Counting objects:  53% (611/1151)        
remote: Counting objects:  54% (622/1151)        
remote: Counting objects:  55% (634/1151)        
remote: Counting objects:  56% (645/1151)        
remote: Counting objects:  57% (657/1151)        
remote: Counting objects:  58% (668/1151)        
remote: Counting objects:  59% (680/1151)        
remote: Counting objects:  60% (691/1151)        
remote: Counting objects:  61% (703/1151)        
remote: Counting objects:  62% (714/1151)        
remote: Counting objects:  63% (726/1151)        
remote: Counting objects:  64% (737/1151)        
remote: Counting objects:  65% (749/1151)        
remote: Counting objects:  66% (760/1151)        
remote: Counting objects:  67% (772/1151)        
remote: Counting objects:  68% (783/1151)        
remote: Counting objects:  69% (795/1151)        
remote: Counting objects:  70% (806/1151)        
remote: Counting objects:  71% (818/1151)        
remote: Counting objects:  72% (829/1151)        
remote: Counting objects:  73% (841/1151)        
remote: Counting objects:  74% (852/1151)        
remote: Counting objects:  75% (864/1151)        
remote: Counting objects:  76% (875/1151)        
remote: Counting objects:  77% (887/1151)        
remote: Counting objects:  78% (898/1151)        
remote: Counting objects:  79% (910/1151)        
remote: Counting objects:  80% (921/1151)        
remote: Counting objects:  81% (933/1151)        
remote: Counting objects:  82% (944/1151)        
remote: Counting objects:  83% (956/1151)        
remote: Counting objects:  84% (967/1151)        
remote: Counting objects:  85% (979/1151)        
remote: Counting objects:  86% (990/1151)        
remote: Counting objects:  87% (1002/1151)        
remote: Counting objects:  88% (1013/1151)        
remote: Counting objects:  89% (1025/1151)        
remote: Counting objects:  90% (1036/1151)        
remote: Counting objects:  91% (1048/1151)        
remote: Counting objects:  92% (1059/1151)        
remote: Counting objects:  93% (1071/1151)        
remote: Counting objects:  94% (1082/1151)        
remote: Counting objects:  95% (1094/1151)        
remote: Counting objects:  96% (1105/1151)        
remote: Counting objects:  97% (1117/1151)        
remote: Counting objects:  98% (1128/1151)        
remote: Counting objects:  99% (1140/1151)        
remote: Counting objects: 100% (1151/1151)        
remote: Counting objects: 100% (1151/1151), done.        
remote: Compressing objects:   0% (1/258)        
remote: Compressing objects:   1% (3/258)        
remote: Compressing objects:   2% (6/258)        
remote: Compressing objects:   3% (8/258)        
remote: Compressing objects:   4% (11/258)        
remote: Compressing objects:   5% (13/258)        
remote: Compressing objects:   6% (16/258)        
remote: Compressing objects:   7% (19/258)        
remote: Compressing objects:   8% (21/258)        
remote: Compressing objects:   9% (24/258)        
remote: Compressing objects:  10% (26/258)        
remote: Compressing objects:  11% (29/258)        
remote: Compressing objects:  12% (31/258)        
remote: Compressing objects:  13% (34/258)        
remote: Compressing objects:  14% (37/258)        
remote: Compressing objects:  15% (39/258)        
remote: Compressing objects:  16% (42/258)        
remote: Compressing objects:  17% (44/258)        
remote: Compressing objects:  18% (47/258)        
remote: Compressing objects:  19% (50/258)        
remote: Compressing objects:  20% (52/258)        
remote: Compressing objects:  21% (55/258)        
remote: Compressing objects:  22% (57/258)        
remote: Compressing objects:  23% (60/258)        
remote: Compressing objects:  24% (62/258)        
remote: Compressing objects:  25% (65/258)        
remote: Compressing objects:  26% (68/258)        
remote: Compressing objects:  27% (70/258)        
remote: Compressing objects:  28% (73/258)        
remote: Compressing objects:  29% (75/258)        
remote: Compressing objects:  30% (78/258)        
remote: Compressing objects:  31% (80/258)        
remote: Compressing objects:  32% (83/258)        
remote: Compressing objects:  33% (86/258)        
remote: Compressing objects:  34% (88/258)        
remote: Compressing objects:  35% (91/258)        
remote: Compressing objects:  36% (93/258)        
remote: Compressing objects:  37% (96/258)        
remote: Compressing objects:  38% (99/258)        
remote: Compressing objects:  39% (101/258)        
remote: Compressing objects:  40% (104/258)        
remote: Compressing objects:  41% (106/258)        
remote: Compressing objects:  42% (109/258)        
remote: Compressing objects:  43% (111/258)        
remote: Compressing objects:  44% (114/258)        
remote: Compressing objects:  45% (117/258)        
remote: Compressing objects:  46% (119/258)        
remote: Compressing objects:  47% (122/258)        
remote: Compressing objects:  48% (124/258)        
remote: Compressing objects:  49% (127/258)        
remote: Compressing objects:  50% (129/258)        
remote: Compressing objects:  51% (132/258)        
remote: Compressing objects:  52% (135/258)        
remote: Compressing objects:  53% (137/258)        
remote: Compressing objects:  54% (140/258)        
remote: Compressing objects:  55% (142/258)        
remote: Compressing objects:  56% (145/258)        
remote: Compressing objects:  57% (148/258)        
remote: Compressing objects:  58% (150/258)        
remote: Compressing objects:  59% (153/258)        
remote: Compressing objects:  60% (155/258)        
remote: Compressing objects:  61% (158/258)        
remote: Compressing objects:  62% (160/258)        
remote: Compressing objects:  63% (163/258)        
remote: Compressing objects:  64% (166/258)        
remote: Compressing objects:  65% (168/258)        
remote: Compressing objects:  66% (171/258)        
remote: Compressing objects:  67% (173/258)        
remote: Compressing objects:  68% (176/258)        
remote: Compressing objects:  69% (179/258)        
remote: Compressing objects:  70% (181/258)        
remote: Compressing objects:  71% (184/258)        
remote: Compressing objects:  72% (186/258)        
remote: Compressing objects:  73% (189/258)        
remote: Compressing objects:  74% (191/258)        
remote: Compressing objects:  75% (194/258)        
remote: Compressing objects:  76% (197/258)        
remote: Compressing objects:  77% (199/258)        
remote: Compressing objects:  78% (202/258)        
remote: Compressing objects:  79% (204/258)        
remote: Compressing objects:  80% (207/258)        
remote: Compressing objects:  81% (209/258)        
remote: Compressing objects:  82% (212/258)        
remote: Compressing objects:  83% (215/258)        
remote: Compressing objects:  84% (217/258)        
remote: Compressing objects:  85% (220/258)        
remote: Compressing objects:  86% (222/258)        
remote: Compressing objects:  87% (225/258)        
remote: Compressing objects:  88% (228/258)        
remote: Compressing objects:  89% (230/258)        
remote: Compressing objects:  90% (233/258)        
remote: Compressing objects:  91% (235/258)        
remote: Compressing objects:  92% (238/258)        
remote: Compressing objects:  93% (240/258)        
remote: Compressing objects:  94% (243/258)        
remote: Compressing objects:  95% (246/258)        
remote: Compressing objects:  96% (248/258)        
remote: Compressing objects:  97% (251/258)        
remote: Compressing objects:  98% (253/258)        
remote: Compressing objects:  99% (256/258)        
remote: Compressing objects: 100% (258/258)        
remote: Compressing objects: 100% (258/258), done.        
warning: 'C:\ProgramData/Git/config' has a dubious owner: '(unknown)'.
For security reasons, it is therefore ignored.
To fix this, please transfer ownership to an admininstrator.
Receiving objects:   0% (1/846)   
Receiving objects:   1% (9/846)   
Receiving objects:   2% (17/846)   
Receiving objects:   3% (26/846)   
Receiving objects:   4% (34/846)   
Receiving objects:   5% (43/846)   
Receiving objects:   6% (51/846)   
Receiving objects:   7% (60/846)   
Receiving objects:   8% (68/846)   
Receiving objects:   9% (77/846)   
Receiving objects:  10% (85/846)   
Receiving objects:  11% (94/846)   
Receiving objects:  12% (102/846)   
Receiving objects:  13% (110/846)   
Receiving objects:  14% (119/846)   
Receiving objects:  15% (127/846)   
Receiving objects:  16% (136/846)   
Receiving objects:  17% (144/846)   
Receiving objects:  18% (153/846)   
Receiving objects:  19% (161/846)   
Receiving objects:  20% (170/846)   
Receiving objects:  21% (178/846)   
Receiving objects:  22% (187/846)   
Receiving objects:  23% (195/846)   
Receiving objects:  24% (204/846)   
Receiving objects:  25% (212/846)   
Receiving objects:  26% (220/846)   
Receiving objects:  27% (229/846)   
Receiving objects:  28% (237/846)   
Receiving objects:  29% (246/846)   
Receiving objects:  30% (254/846)   
Receiving objects:  31% (263/846)   
Receiving objects:  32% (271/846)   
Receiving objects:  33% (280/846)   
Receiving objects:  34% (288/846)   
Receiving objects:  35% (297/846)   
Receiving objects:  36% (305/846)   
Receiving objects:  37% (314/846)   
Receiving objects:  38% (322/846)   
Receiving objects:  39% (330/846)   
Receiving objects:  40% (339/846)   
Receiving objects:  41% (347/846)   
Receiving objects:  42% (356/846)   
Receiving objects:  43% (364/846)   
Receiving objects:  44% (373/846)   
Receiving objects:  45% (381/846)   
Receiving objects:  46% (390/846)   
Receiving objects:  47% (398/846)   
Receiving objects:  48% (407/846)   
Receiving objects:  49% (415/846)   
Receiving objects:  50% (423/846)   
Receiving objects:  51% (432/846)   
Receiving objects:  52% (440/846)   
Receiving objects:  53% (449/846)   
Receiving objects:  54% (457/846)   
Receiving objects:  55% (466/846)   
Receiving objects:  56% (474/846)   
Receiving objects:  57% (483/846)   
Receiving objects:  58% (491/846)   
Receiving objects:  59% (500/846)   
Receiving objects:  60% (508/846)   
Receiving objects:  61% (517/846)   
Receiving objects:  62% (525/846)   
Receiving objects:  63% (533/846)   
Receiving objects:  64% (542/846)   
Receiving objects:  65% (550/846)   
Receiving objects:  66% (559/846)   
Receiving objects:  67% (567/846)   
Receiving objects:  68% (576/846)   
Receiving objects:  69% (584/846)   
Receiving objects:  70% (593/846)   
Receiving objects:  71% (601/846)   
Receiving objects:  72% (610/846)   
Receiving objects:  73% (618/846)   
Receiving objects:  74% (627/846)   
Receiving objects:  75% (635/846)   
Receiving objects:  76% (643/846)   
Receiving objects:  77% (652/846)   
Receiving objects:  78% (660/846)   
Receiving objects:  79% (669/846)   
Receiving objects:  80% (677/846)   
Receiving objects:  81% (686/846)   
Receiving objects:  82% (694/846)   
Receiving objects:  83% (703/846)   
Receiving objects:  84% (711/846)   
Receiving objects:  85% (720/846)   
Receiving objects:  86% (728/846)   
Receiving objects:  87% (737/846)   
Receiving objects:  88% (745/846)   
Receiving objects:  89% (753/846)   
Receiving objects:  90% (762/846)   
Receiving objects:  91% (770/846)   
Receiving objects:  92% (779/846)   
Receiving objects:  93% (787/846)   
remote: Total 846 (delta 714), reused 672 (delta 555), pack-reused 0        
Receiving objects:  94% (796/846)   
Receiving objects:  95% (804/846)   
Receiving objects:  96% (813/846)   
Receiving objects:  97% (821/846)   
Receiving objects:  98% (830/846)   
Receiving objects:  99% (838/846)   
Receiving objects: 100% (846/846)   
Receiving objects: 100% (846/846), 212.03 KiB | 956.00 KiB/s, done.
Resolving deltas:   0% (0/714)   
Resolving deltas:   1% (9/714)   
Resolving deltas:   2% (17/714)   
Resolving deltas:   3% (23/714)   
Resolving deltas:   5% (41/714)   
Resolving deltas:   6% (44/714)   
Resolving deltas:   7% (51/714)   
Resolving deltas:   9% (70/714)   
Resolving deltas:  10% (73/714)   
Resolving deltas:  12% (87/714)   
Resolving deltas:  13% (94/714)   
Resolving deltas:  14% (105/714)   
Resolving deltas:  15% (111/714)   
Resolving deltas:  16% (115/714)   
Resolving deltas:  17% (122/714)   
Resolving deltas:  18% (129/714)   
Resolving deltas:  19% (136/714)   
Resolving deltas:  20% (147/714)   
Resolving deltas:  21% (150/714)   
Resolving deltas:  22% (160/714)   
Resolving deltas:  23% (169/714)   
Resolving deltas:  28% (206/714)   
Resolving deltas:  30% (221/714)   
Resolving deltas:  31% (223/714)   
Resolving deltas:  32% (229/714)   
Resolving deltas:  33% (236/714)   
Resolving deltas:  34% (243/714)   
Resolving deltas:  35% (251/714)   
Resolving deltas:  36% (259/714)   
Resolving deltas:  37% (267/714)   
Resolving deltas:  38% (274/714)   
Resolving deltas:  39% (279/714)   
Resolving deltas:  40% (287/714)   
Resolving deltas:  41% (296/714)   
Resolving deltas:  42% (303/714)   
Resolving deltas:  43% (310/714)   
Resolving deltas:  44% (317/714)   
Resolving deltas:  45% (322/714)   
Resolving deltas:  46% (334/714)   
Resolving deltas:  47% (338/714)   
Resolving deltas:  48% (343/714)   
Resolving deltas:  49% (352/714)   
Resolving deltas:  50% (359/714)   
Resolving deltas:  51% (365/714)   
Resolving deltas:  52% (372/714)   
Resolving deltas:  53% (382/714)   
Resolving deltas:  54% (387/714)   
Resolving deltas:  55% (395/714)   
Resolving deltas:  56% (400/714)   
Resolving deltas:  57% (407/714)   
Resolving deltas:  58% (416/714)   
Resolving deltas:  59% (423/714)   
Resolving deltas:  60% (433/714)   
Resolving deltas:  61% (441/714)   
Resolving deltas:  62% (443/714)   
Resolving deltas:  63% (454/714)   
Resolving deltas:  64% (457/714)   
Resolving deltas:  65% (465/714)   
Resolving deltas:  66% (472/714)   
Resolving deltas:  67% (479/714)   
Resolving deltas:  68% (490/714)   
Resolving deltas:  69% (493/714)   
Resolving deltas:  70% (501/714)   
Resolving deltas:  71% (508/714)   
Resolving deltas:  72% (515/714)   
Resolving deltas:  73% (522/714)   
Resolving deltas:  74% (529/714)   
Resolving deltas:  75% (541/714)   
Resolving deltas:  76% (547/714)   
Resolving deltas:  80% (578/714)   
Resolving deltas:  81% (583/714)   
Resolving deltas:  82% (587/714)   
Resolving deltas:  83% (593/714)   
Resolving deltas:  84% (604/714)   
Resolving deltas:  85% (607/714)   
Resolving deltas:  86% (615/714)   
Resolving deltas:  87% (622/714)   
Resolving deltas:  88% (632/714)   
Resolving deltas:  89% (636/714)   
Resolving deltas:  90% (643/714)   
Resolving deltas:  92% (661/714)   
Resolving deltas:  95% (681/714)   
Resolving deltas:  98% (703/714)   
Resolving deltas:  99% (707/714)   
Resolving deltas: 100% (714/714)   
Resolving deltas: 100% (714/714), completed with 229 local objects.
error: unable to write sha1 filename .git/objects/pack/pack-e95948d388c65d07c5ecdea32781fe0430ef9e5a.idx: Permission denied
fatal: cannot store index file
fatal: index-pack failed
warning: 'C:\ProgramData/Git/config' has a dubious owner: '(unknown)'.
For security reasons, it is therefore ignored.
To fix this, please transfer ownership to an admininstrator.
```

**Comment by niik** [comment]
Hi @lachlan-00, thanks for opening this! The warning message you're seeing about transferring ownership is related to a security vulnerability that the Git for Windows project has fixed, see #8101 and [CVE-2019-1211](https://portal.msrc.microsoft.com/en-US/security-guidance/advisory/CVE-2019-1211) for more information on that.

While I would not have expected the warning message to appear when the system level config file didn't exist (Is that intended behavior @dscho?) I don't believe that it's the reason you're unable to push or pull.

The presence of the warning text simply means that Git for Windows did not deem it safe to read the system level config file and therefore ignores it. Since the file doesn't exist on your system to begin with it should make no difference that it got ignored. Looking at your log there's a fatal error related to not being able to write a pack file which is the reason the fetch failed.

```
error: unable to write sha1 filename .git/objects/pack/pack-e95948d388c65d07c5ecdea32781fe0430ef9e5a.idx: Permission denied
fatal: cannot store index file
fatal: index-pack failed
```

Do you know of any reason why Git wouldn't be permitted to write that file? Do you have any anti-virus software installed on your machine that might be interfering?

Does this error show up for just one repository or multiple?

**Comment by lachlan-00** [comment]
good question! i'll double check the AV tomorrow during work. this has only happened after
this is a work laptop so maybe the installed macafee is blocking this. i didn't check AV as it hasn't come up before this version

**Comment by dscho** [comment]
> I would not have expected the warning message to appear when the system level config file doesn't even exist (Is that intended behavior @dscho?)

Oops. That is a bug, indeed. It should _not_ warn in that case. Sorry about that. I am currently on vacation, so I won't be as fast as usual to fix this :-(

**Comment by lachlan-00** [comment]
What's also interesting is you get some decent errors when you make that folder as well. It breaks the whole app just by making an empty folder

I'll post more logs and check AV in a few hours.

**Comment by lachlan-00** [comment]
can confirm AV is containing the exe to disallow access to the local folder.

when you make the progrmdata directory it will break the ui completely.
![image](https://user-images.githubusercontent.com/1305249/63131276-73fcef80-c000-11e9-9c34-81d576933ac7.png)

deleting the folder and restarting will fix

**Comment by Nicholas-Thompson-fourth** [comment]
I started getting this exact same error immediately after upgrading to GitHub Desktop 2.1.1 yesterday too.

I cannot perform pull operations without getting this error and cannot continue (have not tried to push anything yet).

Likewise I do not have this folder in my system and have no idea what it does or doesn't like about my setup, the error message is useless at describing what is wrong. I have administrator privileges on this machine and get the same error using the "Run as Administrator" option.

Can I rollback to 2.1.0 until this is patched or a solution/work-around is figured out?

![github error](https://user-images.githubusercontent.com/9198273/63156903-a82add00-c00d-11e9-9dd5-13d03d90ca75.png)

**Comment by niik** [comment]
@Nicholas-Thompson-fourth, could you please submit a copy of your latest log file for us to look at?

And additionally, could you scroll to the bottom of that error dialog just so that we can make sure there's not an error being hidden by the warning messages. The ownership warning _should_ not interfere with normal operations, it's only a cautionary message.

**Comment by Nicholas-Thompson-fourth** [comment]
I may have resolved it by installing an older 2012 version, synch-ing with the master in that version and then re-installing the current 2.1.1 version again.

Perhaps not helpful for resolving this issue, but but needed to get myself working again.

Nothing new to pull at the moment to see if the issue is still there, but if it happen again in the future I'll be back to provide more details.

**Comment by lachlan-00** [comment]
My AV has stopped blocking the git exe now after a few days of scanning/updates so only the config folder bug remains

**Comment by niik** [comment]
Thanks for all your input! We believe the issue with the overzealous warning messages is resolved in the latest version of GitHub Desktop which shipped today (2.1.2).
